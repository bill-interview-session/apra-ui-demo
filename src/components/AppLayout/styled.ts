'use client'

import styled from 'styled-components'
import mediaQuery from 'styled-media-query'

export const Wrapper = styled.div`
  height: 100vh;
  display: flex;
  flex-direction: column;
`

export const Header = styled.div`
  width: 100%;
  padding: .5rem 1rem;
  display: flex;
  background-color: var(--color-bg-header);
  color: var(--color);
  font-weight: bolder;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.03), 0 1px 6px -1px rgba(0, 0, 0, 0.02), 0 2px 4px 0 rgba(0, 0, 0, 0.02);
  z-index: 100;

  ${mediaQuery.lessThan('medium')`
    justify-content: center;
  `}
`

export const MainWrapper = styled.div`
  flex: 1;
  padding: 1rem;
  overflow-y: auto;
  background-color: var(--color-bg);

  .pageWrapper {
    padding: 1rem;
    background-color: var(--color-bg-page);
    border-radius: 4px;
  }

  ${
    mediaQuery.lessThan('medium')`
      padding-bottom: 30px;
  `}
`
