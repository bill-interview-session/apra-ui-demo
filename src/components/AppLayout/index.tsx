import { PropsWithChildren } from "react"

import { Header, MainWrapper, Wrapper } from "./styled"

type AppLayoutProps = PropsWithChildren

const AppLayout = ({ children }: AppLayoutProps) => {
  return (
    <Wrapper>
      <Header>Demo App</Header>
      <MainWrapper>
        <div className="pageWrapper">{children}</div>
      </MainWrapper>
    </Wrapper>
  )
}

export default AppLayout
