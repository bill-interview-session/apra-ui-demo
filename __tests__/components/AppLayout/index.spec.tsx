import { render } from '@testing-library/react';

import AppLayout from '@components/AppLayout';

describe('AppLayout component', () => {
  it('should render snapshot', () => {
    const { container } = render(
      <AppLayout>
        <div className="testing" />
      </AppLayout>
    );
    expect(container).toMatchSnapshot();
  });
});
