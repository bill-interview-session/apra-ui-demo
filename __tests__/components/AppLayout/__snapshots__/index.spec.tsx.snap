// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout component should render snapshot 1`] = `
<div>
  <div
    class="sc-77387f8d-0 bDZmIX"
  >
    <div
      class="sc-77387f8d-1 jBaltK"
    >
      Demo App
    </div>
    <div
      class="sc-77387f8d-2 dcRQJc"
    >
      <div
        class="pageWrapper"
      >
        <div
          class="testing"
        />
      </div>
    </div>
  </div>
</div>
`;
